public/js/index.js

```diff
@@ -2810,21 +2810,27 @@ function deleteDependencyModal(id, type) {
 
 function initIssueList() {
     var repolink = $('#repolink').val();
-    $('.new-dependency-drop-list')
+    $('#new-dependency-drop-list')
         .dropdown({
             apiSettings: {
                 url: suburl + '/api/v1/repos/' + repolink + '/issues?q={query}',
                 onResponse: function(response) {
                     var filteredResponse = {'success': true, 'results': []};
+                    var currIssueId = $('#new-dependency-drop-list').data('issue-id');
                     // Parse the response from the api to work with our dropdown
                     $.each(response, function(index, issue) {
+                        // Don't list current issue in the dependency list.
+                        if(issue.id === currIssueId) {
+                            return;
+                        }
                         filteredResponse.results.push({
                             'name'  : '#' + issue.number + '&nbsp;' + htmlEncode(issue.title),
                             'value' : issue.id
                         });
                     });
                     return filteredResponse;
                 },
+                cache: false,
```

Comment:
Why disable cache?

Reply:
Let me explain this with an example: 
Suppose you have two issues A and B. You visit issue A and open its dependency list. It will show up only issue B. Now go to issue B and open its dependency list you will see only issue B listed itself because it will show results from previous cache. Now mark issue B as closed and again open the dependency list for A, it will still show issue B even though it has been closed. So disabling cache is required for solving the issue and since the API uses paging it wouldn't cause a significant performance hit.

Reply:
I wonder if we should just change the length of the cache so that it will only cache for say a 15s

Reply:
@zeripath I checked the documentation for the plugin but didn't find any such option. Did I miss something?

Reply:
You're right there's no such option or ability. (I mean seriously why don't people put things like this in their frameworks.)

Ok. I'll take another look at this. I'm minded to ACK